boards/BMS/Src/App/App_Accumulator.c

```diff
@@ -1,160 +1,89 @@
 #include <stdlib.h>
 #include <assert.h>
-#include <stdbool.h>
 #include "App_Accumulator.h"
 
-struct Accumulator
+enum CellMonitorState
 {
-    ExitCode (*configure_cell_monitors)(void);
-    ExitCode (*read_cell_voltages)(void);
+    CELL_VOLTAGE_STATE = 0U,
+};
 
-    struct InRangeCheck *pack_voltage_in_range_check;
-    struct InRangeCheck *min_cell_voltage_in_range_check;
-    struct InRangeCheck *max_cell_voltage_in_range_check;
-    struct InRangeCheck *average_cell_voltage_in_range_check;
-    struct InRangeCheck *segment_0_voltage_in_range_check;
-    struct InRangeCheck *segment_1_voltage_in_range_check;
-    struct InRangeCheck *segment_2_voltage_in_range_check;
-    struct InRangeCheck *segment_3_voltage_in_range_check;
-    struct InRangeCheck *segment_4_voltage_in_range_check;
-    struct InRangeCheck *segment_5_voltage_in_range_check;
+struct Accumulator
+{
+    // Configure the cell monitoring chip
+    bool (*config_monitoring_chip)(void);
+
+    // Cell voltage monitoring functions
+    bool (*start_cell_voltage_conv)(void);
+    bool (*read_cell_voltages)(void);
+    void (*get_min_cell_v_loc)(uint8_t *, uint8_t *);
+    void (*get_max_cell_v_loc)(uint8_t *, uint8_t *);
+    float (*get_min_cell_voltage)(void);
+    float (*get_max_cell_voltage)(void);
+    float (*get_segment_voltage)(AccumulatorSegments_E);
+    float (*get_pack_voltage)(void);
+    float (*get_avg_voltage)(void);
 };
 
 struct Accumulator *App_Accumulator_Create(
-    ExitCode (*configure_cell_monitors)(void),
-    ExitCode (*read_cell_voltages)(void),
+    bool (*configure_monitoring_chip)(void),
```

Comment:
parameter name `configure_monitoring_chip` should match declaration in header file, could be more explicit that it's the LTC chip in particular

Reply:
ah yeah need to change this in the struct as well. i feel like running misra checks will help us a lot with detecting these. we should look into maybe doing this at some point

Reply:
i would prefer to keep sensor information abstracted from these fxn ptrs though, since the app layer shouldnt really have sensor specific information. Rest of the app code for the vehicle follows this convention so it think its also nice to keep things consistent

Taken from the FSM for example, specific sensor information is abstracted here.
![image](https://user-images.githubusercontent.com/32230980/158759519-e927a3f9-933f-4b0c-8687-c1372936fa3b.png)